### DISCUSSION TITLE
CollectionViewCell内にCollectionViewは配置できるのか

### DISCUSSION THREAD

**Discussion by mahirooo** [lit-iphone-course/lit-iphone-course#48] [Question-original] [answered]
(1) 実現したいこと
ViewController内にあるCollectionViewのCell内に、CollectionViewを配置したい。
(ここで配置したいCollectionViewの用途としては、スタンプの画像を表示させるためです)

(2) 実現するために自分が試した内容とその結果
CollectionViewCell内にCollectionViewを配置して、その中にCollectionViewCellを配置しようとした
→Storyboard上でCollectionViewCellが配置できなかった

(3) 調べた内容や資料
https://qiita.com/Erica_pon/items/f1c6a06e399723f32549
https://zenn.dev/yummy_potato/articles/da11c7a1d27860

CollectionIViewCell内にCollectionViewを配置している記事は見つけられませんでした....
他に良い方法があれば教えていただきたいです！！

(4)GitHubのリンク
https://github.com/mahirooo/TampStamp

**Comment by txxxxc** [comment]
CollectionViewCell内にCollectionViewを配置することは可能です！ですがおそらくコードで実装する必要がありそうです！
miroのデザインを見た感じ、大枠をTableViewを作ってCollectionViewでスタンプの画像を表示する流れで実装可能だと思うので、個人的にはそっちの方法をおすすめします！調べてもらった資料もTableViewにCollectionViewを配置していて、記事通りにやれば動きそうなので！
一旦その方針で実装を進めてもらっても大丈夫そうでしょうか...?

**Reply by mahirooo** [reply]
承知です〜！！ありがとうございます！！

**Reply by mahirooo** [reply]
https://zenn.dev/yummy_potato/articles/da11c7a1d27860

大枠のTableViewは作り終わっていて、上記の記事を参考にTableView内にCollectionViewを配置しようとしているのですが、記事の内容がイマイチ理解できず.....

TableViewCell内にCollectionViewは配置できても、その中にCollectionViewCellが配置できなかったりと行き詰まってしまっているので、少しだけ噛み砕いて説明していただくことは可能でしょうか....！

**Comment by txxxxc** [comment, marked as answer]
[調べた感じ](https://stackoverflow.com/questions/38285469/can%C2%B4t-add-items-to-uicollectionview-inside-uiview-xib)、TableViewCellの中に配置されているCollectionViewにCellを追加することはできなさそうです！！(知りませんでしたごめんなさい🙏マジで申し訳ないです...)

ちょっと大変だけどコード上であればTableViewCellの中に配置されているCollectionViewにCellを登録する事ができるみたいなので、下の記事を参考にしてやってみてもらうことはできそうでしょうか...?
[【Swift】TableViewのなかにCollectionViewを入れる - Qiita](https://qiita.com/REON/items/beb1e94fc0a5c9e25cce)

**Reply by mahirooo** [reply]
了解です〜一旦頑張ってみます！

**Reply by mahirooo** [reply]
Qiitaを参考にしつつTableView内にCollectionViewを配置してみたのですが、エラーが出てしまいビルドできない状態です

エラー
`Redundant conformance of 'TableViewCell' to protocol 'UICollectionViewDataSource'`
`Command EmitSwiftModule failed with a nonzero exit code`

Git最新の状態にしてありますのでご確認いただけると嬉しいです！

**Reply by txxxxc** [reply]
TableViewCell.swiftの
`class TableViewCell: UITableViewCell, UICollectionViewDelegate, UICollectionViewDataSource {`
の部分から`UICollectionViewDataSource`の部分をけすとうまくいくと思います！

UICollectionViewDataSourceを二回継承しようとしているので怒られてっぽいです！

**Reply by mahirooo** [reply]
行けました！！ありがとうございます！！
